modules/core/src/main/java/org/apache/ignite/internal/client/thin/PayloadOutputChannel.java

```diff
@@ -57,6 +63,7 @@ public BinaryOutputStream out() {
 
     /** {@inheritDoc} */
     @Override public void close() {
-        out.close();
+        if (closed.compareAndSet(false, true))
+            out.release();
```

Comment:
The chunk is reusable and may be already acquired for another request. If we release it again, it may be then acquired again and overwritten, causing data corruption.

Reply:
Okay, this makes sense then.